flink-core/src/main/java/org/apache/flink/configuration/ExecutionOptions.java

```diff
@@ -181,4 +182,73 @@ public class ExecutionOptions {
+                                    + " Note: This is an experimental feature under evaluation.");
+
+    /**
+     * The max limit of in-flight records number in async execution mode, 'in-flight' refers to the
+     * records that have entered the operator but have not yet been processed and emitted to the
+     * downstream. If the in-flight records number exceeds the limit, the newly records entering
+     * will be blocked until the in-flight records number drops below the limit.
+     */
+    @Experimental
+    public static final ConfigOption<Integer> ASYNC_INFLIGHT_RECORDS_LIMIT =
+            ConfigOptions.key("execution.async-mode.in-flight-records-limit")
+                    .intType()
+                    .defaultValue(6000)
+                    .withDescription(
+                            "The max limit of in-flight records number in async execution mode, 'in-flight' refers"
+                                    + " to the records that have entered the operator but have not yet been processed and"
+                                    + " emitted to the downstream. If the in-flight records number exceeds the limit,"
+                                    + " the newly records entering will be blocked until the in-flight records number drops below the limit.");
+
+    /**
+     * The size of buffer under async execution mode. Async execution mode provides a buffer
+     * mechanism to reduce state access. When the number of state requests in the buffer exceeds the
+     * batch size, a batched state execution would be triggered. Larger batch sizes will bring
+     * higher end-to-end latency, this option works with {@link #ASYNC_BUFFER_TIMEOUT} to control
+     * the frequency of triggering.
+     */
+    @Experimental
+    public static final ConfigOption<Integer> ASYNC_BUFFER_SIZE =
+            ConfigOptions.key("execution.async-mode.buffer-size")
+                    .intType()
+                    .defaultValue(1000)
+                    .withDescription(
+                            "The size of buffer under async execution mode. Async execution mode provides a buffer mechanism to reduce state access."
+                                    + " When the number of state requests in the active buffer exceeds the batch size,"
+                                    + " a batched state execution would be triggered. Larger batch sizes will bring higher end-to-end latency,"
+                                    + " this option works with 'execution.async-state.buffer-timeout' to control the frequency of triggering.");
+
+    /**
+     * The timeout of buffer triggering in milliseconds. If the buffer has not reached the {@link
+     * #ASYNC_BUFFER_SIZE} within 'buffer-timeout' milliseconds, a trigger will perform actively.
+     */
+    @Experimental
+    public static final ConfigOption<Integer> ASYNC_BUFFER_TIMEOUT =
+            ConfigOptions.key("execution.async-state.buffer-timeout")
```

Comment:
It might be better to change the `execution.async-mode` above to `execution.async-state`.

Reply:
Unify those options to `execution.async-mode`, cause we also provide `Record-order` mode to preserve the order of records that without state access.

Reply:
Got it. I agree with it that `async-mode` is better than `async-state` here, but a single word "buffer" might still not be enough, as a new developer who is not familiar with FLIP-425 might regard this configuration as some buffer related to stream records, and then get confused with this config and `in-flight-records-limit`. A possibly better name that comes to my mind is `execution.async-mode.state-buffer-timeout`. WDYT?

Reply:
Good point👍, changed it to `execution.async-mode.state-buffer-timeout`.

Reply:
Sry guys, I'm afraid I prefer the original proposal of `async-state` in FLIP, since this is only for the stateful operator and even the problem of `record-order` is caused by state async processing...